runtime/core/tensor_layout.h

```diff
@@ -10,55 +10,47 @@
 
 #include <executorch/runtime/core/exec_aten/exec_aten.h>
 #include <executorch/runtime/core/exec_aten/util/scalar_type_util.h>
+#include <executorch/runtime/core/result.h>
 #include <executorch/runtime/core/span.h>
 
 namespace executorch {
 namespace runtime {
 
-namespace {
-size_t calculate_nbytes(
-    const Span<const int32_t>& sizes,
-    const exec_aten::ScalarType& scalar_type) {
-  ssize_t n = 1;
-  for (ssize_t i = 0; i < sizes.size(); i++) {
-    ET_CHECK(sizes[i] >= 0);
-    n *= sizes[i];
-  }
-  // Use the full namespace to disambiguate from c10::elementSize.
-  return n * executorch::runtime::elementSize(scalar_type);
-}
-} // namespace
-
 /**
- * Metadata describing the layout of external tensors (tensors that are not
- stored in the PTE file).
- *
- * The NamedDataMap used to create the TensorLayout must outlive the
- TensorLayout.
+ * Describes the layout of a tensor.
  */
-class TensorLayout {
+class ET_EXPERIMENTAL TensorLayout final {
  public:
-  TensorLayout(
-      executorch::aten::ScalarType scalar_type,
-      Span<const int32_t> sizes,
-      Span<const uint8_t> dim_order)
-      : sizes_(sizes),
-        dim_order_(dim_order),
-        scalar_type_(scalar_type),
-        nbytes_(calculate_nbytes(sizes_, scalar_type_)) {}
+  TensorLayout() = delete;
```

Comment:
I don't think you need to do this since you declare a non-default ctor.

https://en.cppreference.com/w/cpp/language/default_constructor
> ### Implicitly-declared default constructor
> If there is no user-declared constructor or constructor template for a class type, the compiler will implicitly declare a default constructor as an inline public member of its class.